pallets/identity/src/lib.rs

```diff
@@ -901,6 +895,48 @@ decl_module! {
             <RevokeOffChainAuthorization<T>>::insert((signer, auth), true);
             Ok(())
         }
+
+        /// Add `Claim::InvestorUniqueness` claim for a given target identity.
+        ///
+        /// # <weight>
+        ///  Weight of the this extrinsic is depend on the computation that used to validate
+        ///  the proof of claim and that will be constant and not depend on the user inputs.
+        /// # </weight>
+        ///
+        /// # Arguments
+        /// * origin - Who provides the claim to the user, In this case user's account id as user provide
+        /// this claim to itself.
+        /// * target - IdentityId whom claim get assigned.
+        /// * claim - `InvestorUniqueness` claim details.
+        /// * proof - To validate the self attestation.
+        /// * expiry - Expiry of claim.
+        ///
+        /// # Errors
+        /// * `DidMustAlreadyExist` Target should already been a part of the ecosystem.
+        /// * `ClaimVariantNotAllowed` When origin trying to pass claim variant other than `InvestorUniqueness`.
+        /// * `ConfidentialScopeClaimNotAllowed` When issuer is different from target or CDD_ID is invalid for given user.
+        /// * `InvalidScopeClaim1 When proof is invalid.
+        #[weight = 7_500_000_000]
+        pub fn add_investor_uniqueness_claim(origin, target: IdentityId, claim: Claim, proof: InvestorZKProofData, expiry: Option<T::Moment>) -> DispatchResult {
+            // Validate whether the origin is signed or not. Also validate whether the given IdentityId (i.e target) has
+            // already been in the system or not.
+            let issuer = Self::ensure_signer_and_validate_claim_target(origin, target)?;
+
+            // Validate proof and add claim only when the claim variant is `InvestorUniqueness` only
+            // otherwise throw and error.
+            match &claim {
```

Comment:
`if let` would do here.

Reply:
Does `if let` has more benefit over `match`? IMO `match` is much cleaner.

Reply:
If there's no `else` / `_` branch then `if let` reduces 1 level of indentation and lets you omit the `_ => ...` arm, which improves readability, so I'd say always go for it.

The case for `if let ... else` is less clear. I usually try both to see which one rustfmt wants to format more nicely (e.g., will the 1 indent level make a difference in terms of keeping everything on one line?) according to my own subjective taste.

However, there's a good case for `match` and avoiding `_` when its critical that you be notified of more variants by compile errors (often used in `rustc` to avoid surprises at runtime).